pkg/build/strategies/sti/sti.go

```diff
@@ -490,16 +496,25 @@ func (b *STI) Execute(command string, user string, config *api.Config) error {
 			}
 			return nil
 		}
+	} else {
+		close(injectionComplete)
 	}
 
 	if !config.LayeredBuild {
 		wg := sync.WaitGroup{}
 		wg.Add(1)
 		uploadDir := filepath.Join(config.WorkingDir, "upload")
-
 		// TODO: be able to pass a stream directly to the Docker build to avoid the double temp hit
 		r, w := io.Pipe()
 		go func() {
+			// Wait for the injections to complete and check the error. Do not start
+			// streaming the sources when the injection failed.
+			<-injectionComplete
+			if injectionError != nil {
+				wg.Done()
+				return
+			}
+			glog.V(2).Infof("starting the source uploading ...")
```

Comment:
fyi, I want to leave this here for the future debugging... the timing for streaming into container is important as when we fail in OnStart the container is removed.